### DISCUSSION TITLE
Make parking lot fix backwards compatible

### DISCUSSION THREAD

**Discussion by jmaas-onoffice** [onOffice-Web-Org/oo-wp-plugin#354] [_Discarded pitches]
> Time-tracking ticket: T#3270069

## The current fix requries template changes
The fix in #105 returns structured data for the field `multiParkingLot`, which means that the data needs to be rendered by the template. Previous templates do not handle this field correctly, which means that if you want to add the field, you need to update the template.

Since we know about many existing customers who want to display the field, this would mean a lot of work for us.

However, we want to keep the control of how the field is rendered in the template so that users have access to all the information and can personalize it.

I propose the following as a 1-week project.

## Output as plain text in loop
The templates output the fields in a loop:

https://github.com/onOffice-Web-Org/oo-wp-plugin/blob/5f95ee239c457257d9f759f1a69279c75dcc8c56/templates.dist/estate/default.php#L100-L115

In the loop, the `$value` should be set to the [plain text display of the parking lots](https://github.com/onOffice-Web-Org/oo-wp-plugin/blob/e53a7595cb267350087b3a430c6cab81a5d0b503/templates.dist/fields.php#L247).

This allows adding the field to existing websites, without modifying the templates.

## Output as rich object from function
The new template [checks for the parking lot field](https://github.com/onOffice-Web-Org/oo-wp-plugin/blob/5f95ee239c457257d9f759f1a69279c75dcc8c56/templates.dist/estate/default.php#L110) and then [uses the function `get_raw_value()`](https://github.com/onOffice-Web-Org/oo-wp-plugin/blob/5f95ee239c457257d9f759f1a69279c75dcc8c56/templates.dist/estate/parkingLot/ParkingLot.php#L16) to access the field's data. That function should return the full object.

This allows the new template to function and gives more control over the display in the template.

**Comment by jmaas-onoffice** [comment]
@dai-eastgate @LongTrong-exe How complex do you think this change is? Can we do something that achieves this result in 1 week?

**Reply by dai-eastgate** [reply]
> @dai-eastgate @LongTrong-exe How complex do you think this change is? Can we do something that achieves this result in 1 week?

I think it a slightly complicated but I will complete it in 1 week.

**Reply by jmaas-onoffice** [reply]
Don't start working on it yet! Focus on the bets for this cycle. ;)

What are the most important complications that you see? Maybe we can find a way to deal with those first.
